src/main/kotlin/org/dropProject/controllers/StudentAPIController.kt

```diff
@@ -108,4 +108,24 @@ class StudentAPIController(
             return ResponseEntity.ok().body(report)
         }
     }
+
+    @RequestMapping(value = ["/assignment/{assignmentId}/instructions"], method = [(RequestMethod.GET)], produces = [MediaType.APPLICATION_JSON_VALUE])
+    @JsonView(JSONViews.StudentAPI::class)  // to publish only certain fields of the Assignment
+    @ApiOperation(value = "Get the instructions associated with this assignment")
+    fun getInstructionsFragment(@PathVariable assignmentId: String) : ResponseEntity<String> {
+
+        val assignment = assignmentRepository.findById(assignmentId).orElse(null)
+
+        var instructions = ""
+
+        if (assignment != null) {
+
+            instructions = reportService.assignmentTeacherFiles.getHtmlInstructionsFragment(assignment)
```

Comment:
Instead of accessing the `assignmentTeacherFiles` service through the `reportService`, it would be better if you added a dependency for this service in the StudentAPIController constructor. Then, you could use it directly.